### ISSUE TITLE
FEAT: add a PCAReporter

### ISSUE THREAD

**Issue by sylvaincom** [probabl-ai/skore#996]
### Is your feature request related to a problem? Please describe.

As a data scientist, I want to automatically get insights when performing the key machine learning task of [PCA](https://scikit-learn.org/1.5/modules/decomposition.html#pca).

PCA is a dimensionality reduction transformer. It can also be used to perform EDA on your data: by looking at each samples along some principal components etc, you can get insights on your data.

### Describe the solution you'd like

Get inspiration from R's [`FactoMineR`](http://factominer.free.fr/factomethods/analyse-en-composantes-principales.html) which provides several visualizations when fitting a PCA.

### Describe alternatives you've considered, if relevant

-

### Additional context

-

**Comment by glemaitre** [comment]
For me this feature is just a plot to be added in the `ModelReport` or `EstimatorReport` under the `plot` accessor.

**Comment by glemaitre** [comment]
I'm also not a big fan to expose a `transform` method to the reporter because the reporter does not transform data. I would let the user access the `reporter.estimator.transform(X)`.

**Comment by sylvaincom** [comment]
> For me this feature is just a plot to be added in the `ModelReport` or `EstimatorReport` under the `plot` accessor.

Agreed, this issue is waiting on https://github.com/probabl-ai/skore/issues/834
